witness/golang/cmd/witness/internal/witness/witness.go

```diff
@@ -117,48 +115,93 @@ func (w *Witness) GetCheckpoint(logID string) ([]byte, error) {
 	}
 	cosigned, err := note.Sign(n, w.Signer)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("couldn't sign checkpoint: %v", err)
 	}
 	return cosigned, nil
 }
 
 // Update updates the latest checkpoint if chkptRaw is consistent with the current
-// latest one for this log (according to latestSize).   It returns the latest
-// checkpoint size before the update was applied (or just what was fetched if the
-// update was unsuccessful).
-func (w *Witness) Update(ctx context.Context, logID string, latestSize uint64, chkptRaw []byte, proof [][]byte) (uint64, error) {
+// latest one for this log.   It returns the latest checkpoint size before the
+// update was applied (or just what was fetched if the update was unsuccessful).
+func (w *Witness) Update(ctx context.Context, logID string, chkptRaw []byte, proof [][]byte) (uint64, error) {
+	// If we don't witness this log then no point in going further.
+	logInfo, ok := w.Logs[logID]
+	if !ok {
+		return 0, fmt.Errorf("no information for log %q", logID)
+	}
```

Comment:
Maybe use the "not found" notion? E.g. "log %q not found". Potentially use `status` and `codes` packages to return this with `codes.NotFound`.

Reply:
Done!